File: kotlin/euler/src/main/kotlin/com/rbleggi/euler/P22.kt
package com.rbleggi.euler

fun mainP22() {
    val resource = object {}.javaClass.getResourceAsStream("/names.txt")
    val names = resource?.bufferedReader()?.readText()?.replace("\"", "")?.split(",")?.sorted() ?: emptyList()

    fun nameValue(name: String): Int = name.sumOf { it - 'A' + 1 }

    val totalScore = names.mapIndexed { idx, name ->
        nameValue(name) * (idx + 1)
    }.sum()

    println(totalScore)
}

File: scala-3/euler/src/main/scala/com/rbleggi/euler/P22.scala
package com.rbleggi.euler

import scala.io.Source

@main def runP22(): Unit = {
  val resource = getClass.getResourceAsStream("/names.txt")
  val names = Source.fromInputStream(resource).mkString.replaceAll("\"", "").split(",").sorted

  def nameValue(name: String): Int = name.map(_ - 'A' + 1).sum

  val totalScore = names.zipWithIndex.map { case (name, idx) =>
    nameValue(name) * (idx + 1)
  }.sum

  println(totalScore)
}
